src/amo/pages/SearchPage/index.js

```diff
@@ -43,9 +45,23 @@ export class SearchPageBase extends React.Component<InternalProps> {
 
     const { clientApp, filters, lang, location } = props;
 
+    let pathname = '/search/';
     let shouldRedirect = false;
     const newFilters = { ...filters };
 
+    // If this is an old category search, redirect to the new category page.
+    if (newFilters.category && newFilters.addonType) {
+      pathname = getCategoryResultsPathname({
+        addonType: newFilters.addonType,
+        slug: newFilters.category,
+      });
+
+      delete newFilters.addonType;
+      delete newFilters.category;
```

Comment:
I think we should also delete `sort` if it's equal to the default sort because we should redirect to the canonical category url when possible.